reframe/core/modules.py

```diff
@@ -207,6 +208,30 @@ def is_module_loaded(self, name):
     def _is_module_loaded(self, name):
         return self._backend.is_module_loaded(Module(name))
 
+    def load_mapping_from_file(self, module_mapping_file):
+        """Update ``self.module_map`` with a mapping defined in a file."""
+        module_mapping = {}
+        try:
+            with open(module_mapping_file, 'r') as f:
+                for line in f.readlines():
+                    line = re.split(r'\s*#\s*', line.strip())[0]
+                    if not line:
+                        continue
+
+                    try:
+                        key, value = re.split(r'\s*:\s*', line.strip(':'))
+                        values_list = re.findall(
+                            r'(?P<value>[A-Za-z0-9_\-/\+\.]+)', value)
```

Comment:
Regexes are not very useful here. They complicate the code and restrict the syntax (in this particular case). The idea is simple:

1. Split the line at `:` once (with the string's `split()` method)
2. If any of the left or right parts are empty, that's an error (here you can throw a `ConfigError`)
3. Your left part is the key
4. Split the right part at `,`, the resulting list contains your mappings for the key.
5. Set `self.module_map[key] = value` and there you are.

Reply:
When throwing a `ConfigError` to denote problems in the file, it's good to keep track of the line numbers. You can achieve this as follows:

```python
for lineno, line in enumerate(fp):
```